src/Pim/Component/Catalog/Normalizer/Indexing/Product/PropertiesNormalizer.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+namespace Pim\Component\Catalog\Normalizer\Indexing\Product;
+
+use Pim\Component\Catalog\Model\ProductInterface;
+use Pim\Component\Catalog\Normalizer\Standard\Product\PropertiesNormalizer as StandardPropertiesNormalizer;
+use Symfony\Component\Serializer\Normalizer\NormalizerInterface;
+use Symfony\Component\Serializer\Normalizer\SerializerAwareNormalizer;
+
+/**
+ * Transform the properties of a product object (fields and product values)
+ * to the indexing format.
+ *
+ * @author    Julien Janvier <jjanvier@akeneo.com>
+ * @copyright 2017 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+class PropertiesNormalizer extends SerializerAwareNormalizer implements NormalizerInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function normalize($product, $format = null, array $context = [])
+    {
+        if (!$this->serializer instanceof NormalizerInterface) {
+            throw new \LogicException('Serializer must be a normalizer');
+        }
+
+        $data = [];
+        $data[StandardPropertiesNormalizer::FIELD_IDENTIFIER] = $product->getIdentifier();
+        $data[StandardPropertiesNormalizer::FIELD_VALUES] = [];
```

Comment:
Why not directly call here a "chained product value normalizer", which perform a foreach on each value and normalize it with the right normalizer according to its attribute type?